src/common/imagwebp.cpp

```diff
@@ -0,0 +1,231 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        src/common/imagwebp.cpp
+// Purpose:     wxImage WebP handler
+// Author:      Hermann Höhne
+// Copyright:   (c) Hermann Höhne
+// Licence:     wxWidgets licence
+/////////////////////////////////////////////////////////////////////////////
+
+// based on code by Sylvain Bougnoux, Khral Steelforge and Markus Juergens
+// see https://forums.wxwidgets.org/viewtopic.php?t=39212
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#if wxUSE_IMAGE && wxUSE_LIBWEBP
+
+#include "wx/imagwebp.h"
+#include "webp/demux.h"
+#include "webp/decode.h"
+#include "webp/encode.h"
+#include <vector>
+
+#ifndef WX_PRECOMP
+    #include "wx/intl.h"
+    #include "wx/log.h"
+#endif
+
+//-----------------------------------------------------------------------------
+// wxWEBPHandler
+//-----------------------------------------------------------------------------
+
+wxIMPLEMENT_DYNAMIC_CLASS(wxWEBPHandler, wxImageHandler);
+
+#if wxUSE_STREAMS
+
+#include <wx/mstream.h>
+#include <memory>
+#include <functional>
+
+typedef std::unique_ptr<WebPDemuxer, std::function<void(WebPDemuxer*)>> WebPDemuxerPtr;
+
+bool DecodeWebPDataIntoImage(wxImage *image, WebPData *webp_data, bool verbose) {
```

Comment:
And now a formatting nitpick:
```suggestion
bool DecodeWebPDataIntoImage(wxImage *image, WebPData *webp_data, bool verbose)
{
```